Sources/RoktUXHelper/Services/LayoutTransformer/LayoutTransformer.swift

```diff
@@ -460,45 +536,45 @@ struct LayoutTransformer<Expander: PayloadExpander, Extractor: DataExtractor> wh
                                     layoutState: layoutState)
     }
 
-    func getCreativeResponse(responseKey: String,
-                             openLinks: LinkOpenTarget?,
-                             styles: LayoutStyle<CreativeResponseElements,
-                                                 ConditionalStyleTransition<CreativeResponseTransitions, WhenPredicate>>?,
-                             children: [LayoutSchemaViewModel]?,
-                             slot: SlotOfferModel?) throws -> LayoutSchemaViewModel {
-        guard let responseOptionsMap = slot?.offer?.creative.responseOptionsMap,
-              (responseKey == BNFNamespace.CreativeResponseKey.positive.rawValue
-               && responseOptionsMap.positive != nil)
-                ||
-                (responseKey == BNFNamespace.CreativeResponseKey.negative.rawValue
-                 && responseOptionsMap.negative != nil)
-        else {
+    func getCreativeResponse(model: CreativeResponseModel<LayoutSchemaModel, WhenPredicate>,
+                             context: Context) throws -> LayoutSchemaViewModel {
+        guard case let .inner(.generic(offer)) = context, let offer else {
+            throw LayoutTransformerError.InvalidMapping()
+        }
+        var updatedContext: Context
+        if model.responseKey == BNFNamespace.CreativeResponseKey.positive.rawValue, offer.creative.responseOptionsMap?.positive.isEmpty.not == true {
```

Comment:
I can't think of the correct terminology but it's usually a code smell to do do `== true` when the value itself is a boolean. Can this be removed by unwrapping the optional earlier?